### PR TITLE
Load videos by URL

### PR DESCRIPTION
**Pull request by Dhaulagiri**
Addresses #4: 
- Remove old api in favor of basing everything off the url that is
passed in
- Updated tests
- Updated Readme to include usage instructions

### CODE REVIEW

**Comment by Dhaulagiri** [comment]
Thoughts on this @poteto?

**Comment by poteto** [comment]
@Dhaulagiri looks good to me, could you please fix the merge conflict? I'll have a bit of a play with it tonight :)

**Comment by Dhaulagiri** [comment]
Sure thing, I'll try to rebase tonight.

**Comment by Dhaulagiri** [comment]
Let me know if this doesn't work and I will take another look.  The rebase got a bit wonky...

**Comment by poteto** [comment]
@Dhaulagiri Sorry, could you do another empty commit/push/edit the readme to your branch? I just updated some CircleCI settings to run on PRs.

File: tests/unit/lazy-video-providers-test.js [hunk 1]
```diff
@@ -19,28 +19,28 @@ module('Lazy Video Providers', {
   }
 });
 
-test('_getProvider with explicit provider', function() {
+test('_getProvider based on `url`', function() {
   var provider;
 
-  provider = service._getProvider('youtube');
+  provider = service._getProvider('https://www.youtube.com/watch?v=gvdf5n-zI14');
```
Comment (poteto) [inline tests/unit/lazy-video-providers-test.js]: Could we add more test cases, particularly for the optional params YouTube sometimes adds to URLs?

**Comment by Dhaulagiri** [comment]
Fixed the Phantom issue (now I remember that we can run tests using Phantom locally), and I added a few more unit test scenarios.

**Comment by poteto** [comment]
@Dhaulagiri Thanks! :) I'm pulling it down now, and will push a new release pretty soon

**Pull request merged**